packages/asyncapi/src/generator.ts

```diff
@@ -0,0 +1,334 @@
+import type { ContractDefinition } from "@amqp-contract/contract";
+import { match } from "ts-pattern";
+
+/**
+ * AsyncAPI 3.0.0 Specification
+ */
+export interface AsyncAPIDocument {
+  asyncapi: "3.0.0";
+  info: AsyncAPIInfo;
+  servers?: Record<string, AsyncAPIServer>;
+  channels?: Record<string, AsyncAPIChannel>;
+  operations?: Record<string, AsyncAPIOperation>;
+  components?: AsyncAPIComponents;
+}
+
+export interface AsyncAPIInfo {
+  title: string;
+  version: string;
+  description?: string;
+  termsOfService?: string;
+  contact?: {
+    name?: string;
+    url?: string;
+    email?: string;
+  };
+  license?: {
+    name: string;
+    url?: string;
+  };
+}
+
+export interface AsyncAPIServer {
+  host: string;
+  protocol: string;
+  description?: string;
+  variables?: Record<string, { default: string; description?: string }>;
+}
+
+export interface AsyncAPIChannel {
+  address: string;
+  messages?: Record<string, AsyncAPIMessageRef>;
+  description?: string;
+  bindings?: {
+    amqp?: {
+      is: "queue" | "routingKey";
+      queue?: {
+        name: string;
+        durable?: boolean;
+        exclusive?: boolean;
+        autoDelete?: boolean;
+      };
+      exchange?: {
+        name: string;
+        type: "topic" | "direct" | "fanout" | "headers";
+        durable?: boolean;
+        autoDelete?: boolean;
+      };
+    };
+  };
+}
+
+export interface AsyncAPIOperation {
+  action: "send" | "receive";
+  channel: AsyncAPIRef;
+  messages?: AsyncAPIMessageRef[];
+  description?: string;
+}
+
+export interface AsyncAPIMessageRef {
+  $ref?: string;
+  payload?: AsyncAPISchema;
+  contentType?: string;
+  name?: string;
+  title?: string;
+  summary?: string;
+  description?: string;
+}
+
+export interface AsyncAPISchema {
+  type?: string;
+  properties?: Record<string, AsyncAPISchema>;
+  required?: string[];
+  items?: AsyncAPISchema;
+  $ref?: string;
+  description?: string;
+  format?: string;
+  minimum?: number;
+  maximum?: number;
+  minLength?: number;
+  maxLength?: number;
+  pattern?: string;
+  enum?: unknown[];
+}
+
+export interface AsyncAPIRef {
+  $ref: string;
+}
+
+export interface AsyncAPIComponents {
+  messages?: Record<string, AsyncAPIMessageRef>;
+  schemas?: Record<string, AsyncAPISchema>;
+}
+
+/**
+ * Options for generating AsyncAPI specification
+ */
+export interface GenerateAsyncAPIOptions {
+  info: Omit<AsyncAPIInfo, "title" | "version"> & {
+    title?: string;
+    version?: string;
+  };
+  servers?: Record<string, AsyncAPIServer>;
+}
+
+/**
+ * Convert a Zod schema to AsyncAPI schema
+ */
+function zodSchemaToAsyncAPI(schema: unknown): AsyncAPISchema {
```

Comment:
like orpc with @orpc/openapi, @amqp-contract/asyncapi should not know zod

we should have a dedicated zod package with a specific function ZodToJsonSchemaConverter that will be called in this zod agnostic package

then we will have the same pattern for valibot and for arktype